server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -213,7 +225,11 @@ public static IndexTemplateMetaData readFrom(StreamInput in) throws IOException
         Builder builder = new Builder(in.readString());
         builder.order(in.readInt());
         if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1)) {
-            builder.patterns(in.readList(StreamInput::readString));
+            List<String> indexPatterns = in.readList(StreamInput::readString);
+            if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+                builder.autoCreateIndex(Boolean.parseBoolean(indexPatterns.remove(indexPatterns.size() - 1)));
```

Comment:
Adding a serialized `boolean` to the end of this list feels really dirty, but I couldn't get the BwC tests to pass when adding a new byte to the message (even when wrapping writing that boolean in a `if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {`). Maybe I'm making a mistake here?

 ```java
if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {`
  autoCreateIndex = in.readBoolean();
}
```

did not work for me in BwC at least.

Reply:
Maybe this can be resolved in a different way since the issues mentions users will have to update their templates when using the new auto create mode and BwC isn't really an issue?